cmd/controller/main.go

```diff
@@ -335,16 +335,8 @@ func startControllerViaLeaderElection(runOptions controllerRunOptions) error {
 	}
 	go le.Run()
 
-	var g run.Group
-	{
-		g.Add(func() error {
-			<-runOptions.parentCtx.Done()
-			return errors.New("closing the app because leadership position was lost")
-		}, func(err error) {
-			runOptions.parentCtxDone()
-		})
-	}
-	return g.Run()
+	<-runOptions.parentCtx.Done()
```

Comment:
I've just realised that there is no need to use additional group. At some point in the past we had a code that was using a derived context what seemed to be valid approach.